README.md
microfocus chef cookbook
========================

[![Build Status](https://travis-ci.org/universityofderby/chef-microfocus.svg?branch=master)](https://travis-ci.org/universityofderby/chef-microfocus)

The `microfocus` cookbook provides the `microfocus_server_express` resource.
This resource installs Micro Focus Server Express and License Manager, installs the specified license, then enables and starts the License Manager service.

Requirements
------------
- Chef 12.5 or higher
- Ruby 2.0 or higher (preferably from the Chef full-stack installer)
- Network accessible package repositories

Platform Support
----------------
The following platforms have been tested with Test Kitchen:
- centos-7

Dependencies
------------
This cookbook depends on the following cookbooks.
- `ark`
- `systemd`

Usage
-----
Include `microfocus` as a dependency in your cookbook's `metadata.rb`.

```ruby
depends 'microfocus', '~> 4.1.0'
```

Resources
---------
### microfocus_server_express
Define a `microfocus_server_express` resource in your recipe. E.g.

```ruby
microfocus_server_express '5.1wp11' do
  checksum 'ec833c62bdb63f48b7bf7b83b0100e0c82317f9653096d03ba2c9be27a0f6ebd'
  license_number 'license_number'
  serial_number 'serial_number'
  url 'http://artifacts.local.org/microfocus/server-express/sx51_wp11_redhat_x86_64_dev.tar'
end
```

#### Properties
- `checksum` - SHA-256 checksum for the Server Express archive.
- `group` - group for the Server Express directory (default: 'root').
- `install_responses` - array of hashes (pattern to match and input value) to override the default install responses.
- `license_manager_path` - full path to install License Manager (default: '/opt/microfocus/mflmf').
- `license_number` - Server Express license number (required: true).
- `mflmcmd_responses` - array of hashes (pattern to match and input value) to override the default mflmcmd responses.
- `mode` - mode for the Server Express directory (default: 0o755).
- `owner` - owner for the Server Express directory (default: 'root').
- `serial_number` - Server Express serial number (required: true).
- `server_express_path` - full path to install Server Express (default: '/opt/microfocus/cobol').
- `url` - URL for the Server Express archive (required: true).
- `version` - Server Express version (name_property: true).

#### Actions
- `:create` - installs Micro Focus Server Express and License Manager, installs the specified license, then enables and starts the License Manager service.

### microfocus_visual_cobol
Define a `microfocus_visual_cobol` resource in your recipe. E.g.

```ruby
microfocus_visual_cobol 'visual_cobol' do
  visual_cobol_checksum 'Install file checksum'
  visual_cobol_url 'http://artifacts.local.org/artifactory/software/microfocus/visual-cobol/setup_visualcobol_devhub_4.0_patchupdate04_196243_redhat_x86_64'
  visual_cobol_license_checksum 'license file checksum'
  visual_cobol_license_url 'http://artifacts.local.org/artifactory/software/microfocus/visual-cobol/PS-CS-UNIX-Linux.mflic'
end
```

#### Properties
- `group` - group for the Visual Cobol install and license files (default: 'root').
- `mode` - mode for the Visual Cobol install and license files(default: 0o775).
- `owner` - owner for the Visual Cobol install and license files (default: 'root').
- `visual_cobol_checksum` - SHA-256 checksum for the Visual Cobol install file.
- `visual_cobol_install_path` - full path to Visual Cobol (default: '/opt/microfocus/VisualCOBOL').
- `visual_cobol_license_checksum` - SHA-256 checksum for the Visual Cobol license file.
- `visual_cobol_license_bin_path` - full path to Visual Cobol license tool binaries (default: '/var/microfocuslicensing/bin/').
- `visual_cobol_license_path` - full path to install Visual Cobol license file (default: '/opt/microfocus/VisualCOBOL/etc/PS-VC-UNIX-Linux').
- `visual_cobol_license_url` - URL for the Visual Cobol license archive (required: true).
- `visual_cobol_setup_path` - full path to install Visual Cobol application (default: '/tmp/setup_visualcobol').
- `visual_cobol_url` - URL for the Visual Cobol archive (required: true).

#### Actions
- `:create` - installs Micro Focus VisualCOBOL and Visual Cobol License Management Tool, installs the specified license.

Contributing
------------
1. Fork the repository on GitHub.
2. Create a named feature branch (like `add_component_x`).
3. Write your change.
4. Write tests for your change (this cookbook currently uses InSpec with Test Kitchen).
5. Run the tests, ensuring they all pass.
6. Submit a Pull Request using GitHub.

License and Authors
-------------------
Authors:
- Richard Lock
- Conor Kelly

Copyright 2020 University of Derby

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

<http://www.apache.org/licenses/LICENSE-2.0>

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


CHANGELOG.md
microfocus cookbook CHANGELOG
=============================
4.1.0 (2020-08-06)
------------------
- Updates to packages, added glibc-devel in addition to existing glibc-devel.i686.

4.0.0 (2020-08-06)
------------------
- Added mfcesd and lserv services to auto start on reboot.

3.1.0 (2020-08-04)
------------------
- Added microfocus_visual_cobol resource.

3.0.0 (2016-10-05)
------------------
- Updated constraint for ark cookbook to ~> 2.0. version is now name_property. Removed default recipe.

2.1.2 (2016-04-27)
------------------
- Removed leading / from template resources source property.

2.1.1 (2016-04-22)
------------------
- Added Travis CI build status.

2.1.0 (2016-04-20)
------------------
- Updated microfocus_server_express resource to manually install init.d/systemd service.

2.0.1 (2016-04-17)
------------------
- Updated readme.

2.0.0 (2016-04-16)
------------------
- Refactored as resource cookbook.

1.0.0 (2015-06-30)
------------------
- Updated recipe.

0.1.0 (2015-05-10)
------------------
- Initial release.


test/integration/default/default_spec.rb
# packages
%w(gcc glibc libgcc).each do |p|
  describe package(p) do
    it { should be_installed }
  end
end

# microfocus directories
{
  '/opt/microfocus' => '0755',
  '/opt/microfocus/cobol' => '0755',
  '/opt/microfocus/mflmf' => '0700'
}.each do |f, m|
  describe file(f) do
    it { should be_directory }
    its(:owner) { should eq 'root' }
    its(:group) { should eq 'root' }
    its(:mode) { should eq m }
  end
end

# microfocus files
{
  '/opt/microfocus/cobol/install' => '0555',
  '/opt/microfocus/mflmf/mflmcmd' => '0555',
  '/opt/microfocus/mflmf/mflm_manager' => '0544'
}.each do |f, m|
  describe file(f) do
    its(:owner) { should eq 'root' }
    its(:group) { should eq 'root' }
    its(:mode) { should eq m }
  end
end

# license manager startup script
describe file('/etc/mflmrcscript') do
  its(:content) { should match(%r{cd \/opt\/microfocus\/mflmf}) }
  its(:content) { should match(%r{\.\/mflm_manager}) }
  its(:owner) { should eq 'root' }
  its(:group) { should eq 'root' }
  its(:mode) { should eq 0755 }
end

# license manager service
describe service('mflm') do
  it { should be_installed }
  it { should be_running }
end

describe service('mflm').runlevels(2, 3, 4, 5) do
  it { should be_enabled }
end

# license manager process
describe processes('./mflm_manager') do
  its('list.length') { should eq 1 }
  its('users') { should eq ['root'] }
end


test/cookbooks/microfocus_test/metadata.rb
name 'microfocus_test'
version '0.1.0'

depends 'microfocus'


test/integration/visual_cobol/default_spec.rb
# packages
%w[ed gcc glibc glibc-devel glibc-devel.i686 libgcc libstdc++ pax xterm].each do |p|
  describe package(p) do
    it { should be_installed }
  end
end

# microfocus visual cobol directories
[
  '/opt/microfocus/VisualCOBOL/bin/cob',
  '/var/microfocuslicensing/bin/cesadmintool.sh'
].each do |f|
  describe file(f) do
    its(:owner) { should eq 'root' }
    its(:group) { should eq 'root' }
    its(:mode) { should eq 0o555 }
  end
end

%w[lserv mfcesd].each do |p|
  describe processes(p) do
    it { should exist }
  end
end

describe command('/var/microfocuslicensing/bin/lsmon') do
  its('stdout') { should match (/\/var\/microfocuslicensing\/bin\/lservrc.net/) }
  its('stdout') { should match (/License status\s*:\s*Active/) }
end

# license manager and daemon services
%w[lserv mfcesd].each do |s|
  describe service(s) do
    it { should be_enabled }
    it { should be_running }
  end
end


resources/visual_cobol.rb
#
# Cookbook Name:: microfocus
# Resource:: visual_cobol
#
# Copyright 2020 University of Derby
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

resource_name :microfocus_visual_cobol
default_action :create

property :group, String, default: 'root'
property :mode, Integer, default: 0o775
property :owner, String, default: 'root'
property :visual_cobol_checksum, String
property :visual_cobol_install_path, String, default: '/opt/microfocus/VisualCOBOL'
property :visual_cobol_license_checksum, String
property :visual_cobol_license_bin_path, String, default: '/var/microfocuslicensing/bin'
property :visual_cobol_license_path, String, default: lazy {"#{visual_cobol_install_path}/etc/PS-VC-UNIX-Linux"}
property :visual_cobol_license_url, String, required: true
property :visual_cobol_setup_path, String, default: '/tmp/setup_visualcobol'
property :visual_cobol_url, String, required: true

# default action :create
action :create do
  %w[ed gcc glibc glibc-devel glibc-devel.i686 libgcc libstdc++ pax xterm].each do |p|
    package p
  end

  # setup install file location
  remote_file new_resource.visual_cobol_setup_path do
    source new_resource.visual_cobol_url
    checksum new_resource.visual_cobol_checksum
    owner new_resource.owner
    group new_resource.group
    mode new_resource.mode
    action :create
    not_if { ::File.exist?(new_resource.visual_cobol_setup_path) }
  end

  # install visual cobol
  execute 'visual_cobol_install' do
    command "#{new_resource.visual_cobol_setup_path} -silent -IacceptEULA -noplatformcheck"
    not_if { ::File.exist?(::File.join(new_resource.visual_cobol_install_path, 'bin/cob')) }
  end

  # copy license file to target
  remote_file new_resource.visual_cobol_license_path do
    source new_resource.visual_cobol_license_url
    checksum new_resource.visual_cobol_license_checksum
    owner new_resource.owner
    group new_resource.group
    mode new_resource.mode
    action :create
    not_if { ::File.exist?(new_resource.visual_cobol_license_path) }
  end

  # install license
  execute 'visual_cobol_license_install' do
    command "#{new_resource.visual_cobol_license_bin_path}/cesadmintool.sh -install #{new_resource.visual_cobol_license_path}"
    not_if "/var/microfocuslicensing/bin/lsmon | grep '/var/microfocuslicensing/bin/lservrc\.net'"
    notifies :run, 'execute[stop_mfcesd]', :immediately
  end

  # stop mfcesd
  execute 'stop_mfcesd' do
    command "#{::File.join(new_resource.visual_cobol_license_bin_path, 'stopmfcesd.sh')}"
    action :nothing
    notifies :run, 'execute[stop_lserv]', :immediately
  end

  # stop lserv
  execute 'stop_lserv' do
    command "#{::File.join(new_resource.visual_cobol_license_bin_path, 'stoplserv.sh')}"
    action :nothing
  end

  # lserv service
  systemd_unit 'lserv.service' do
    content({
      Unit: {
        After: 'network.target',
        Description: 'Microfocus Visual Cobol lserv service'
      },
      Service: {
        ExecStart: "#{::File.join(new_resource.visual_cobol_license_bin_path, 'startlserv.sh')}",
        ExecStop: "#{::File.join(new_resource.visual_cobol_license_bin_path, 'stoplserv.sh')}",
        Type: 'forking'
      },
      Install: {
        WantedBy: 'multi-user.target'
      }
    })
    action [:create, :enable, :start]
  end

  # mfcesd service
  systemd_unit 'mfcesd.service' do
    content({
      Unit: {
        After: 'network.target lserv.service',
        Description: 'Microfocus Visual Cobol mfcesd service',
        Requires: 'lserv.service'
      },
      Service: {
        ExecStart: "#{::File.join(new_resource.visual_cobol_license_bin_path, 'startmfcesd.sh')}",
        ExecStop: "#{::File.join(new_resource.visual_cobol_license_bin_path, 'stopmfcesd.sh')}",
        Type: 'forking'
      },
      Install: {
        WantedBy: 'multi-user.target'
      }
    })
    action [:create, :enable, :start]
  end
end


resources/server_express.rb
#
# Cookbook Name:: microfocus
# Resource:: server_express
#
# Copyright 2016 University of Derby
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

resource_name :microfocus_server_express
default_action :create

property :checksum, String
property :group, String, default: 'root'
property :license_manager_path, default: '/opt/microfocus/mflmf'
property :license_number, String, required: true
property :mode, Integer, default: 0o755
property :owner, String, default: 'root'
property :serial_number, String, required: true
property :server_express_path, default: '/opt/microfocus/cobol'
property :url, String, required: true
property :version, String, name_property: true
property :install_responses, Array, default: lazy {
  [
    { '\(y\/n\)' => "y\n" }, # 1. Do you wish to continue (y/n):
    { '\(y\/n\)' => "y\n" }, # 2. Do you agree to the terms of the License Agreement? (y/n):
    { '\(y\/n\)' => "y\n" }, # 3. Please confirm that you want to continue with this installation (y/n):
    { 'Please press return when you are ready:' => "\n" }, # 4. Please press return when you are ready:
    { '\(y\/n\)' => "y\n" }, # 5. Please confirm your understanding of the above reference environment details (y/n):
    { '\(y\/n\)' => "n\n" }, # 6. Do you want to make use of COBOL and Java working together? (y/n):
    { '\(y\/n\)' => "y\n" }, # 7. Would you like to install LMF now? (y/n):
    { 'Press Enter for default directory' => "#{license_manager_path}\n" }, # 8. Enter the directory name where you wish to install License Manager
    { '\(y\/n\)' => "y\n" }, # 9. do you wish to create it ? (y/n)
    { '\(y\/n\)' => "y\n" }, # 10. Do you want only superuser to be able to access the License Admin System? (y/n)
    { '\(y\/n\)' => "n\n" }, # 11. Do you want license manager to be automatically started at boot time? (y/n)
    { 'Please enter either 32 or 64 to set the system default mode:' => "64\n" }, # 12. Please enter either 32 or 64 to set the system default mode:
    { '\(y\/n\)' => "n\n" } # 13. Do you wish to configure Enterprise Server now? (y/n):
  ]
}
property :mflmcmd_responses, Array, default: lazy {
  [
    { 'License' => "I\n" }, # 1. Select the function you require from the list:
    { 'Serial Number' => "#{serial_number}\n" }, # 2. Enter the Serial Number part of the License Key:
    { 'License Number' => "#{license_number}\n" } # 3. Enter the License Number part of the License Key:
  ]
}

# default action :create
# greenletters gem required for responding to interactive install and mflmcmd commands
action :create do
  chef_gem 'greenletters' do
    compile_time true
  end
  require 'greenletters'

  # install required packages
  %w[gcc glibc glibc.i686 libgcc libgcc.i686].each do |p|
    package p
  end

  # create base directory
  directory ::File.basename(server_express_path) do
    owner new_resource.owner
    group new_resource.group
    mode new_resource.mode
    recursive true
  end

  # extract server express archive
  ark ::File.basename(server_express_path) do
    path ::File.dirname(server_express_path)
    url new_resource.url
    checksum new_resource.checksum if property_is_set?(:checksum)
    owner new_resource.owner
    group new_resource.group
    mode new_resource.mode
    strip_components 0
    not_if { ::File.exist?(server_express_path) }
    notifies :run, 'ruby_block[install]', :immediately
    notifies :run, 'ruby_block[mflmcmd]', :immediately
    action :put
  end

  # execute server express install
  ruby_block 'install' do
    block do
      install = Greenletters::Process.new(::File.join(server_express_path, 'install'), transcript: $stdout, timeout: 300)
      install.on(:output, /--more--/i) do
        install << ' '
      end
      install.start!
      install_responses.each do |r|
        r.each do |p, i|
          install.wait_for(:output, /#{p}/i)
          install << i
        end
      end
      install.wait_for(:exit)
    end
    only_if { ::File.exist?(::File.join(server_express_path, 'install')) }
    action :nothing
  end

  # execute mflmcmd to install license
  ruby_block 'mflmcmd' do
    block do
      mflmcmd = Greenletters::Process.new("cd #{license_manager_path};./mflmcmd", transcript: $stdout, timeout: 300)
      mflmcmd.start!
      mflmcmd_responses.each do |r|
        r.each do |p, i|
          mflmcmd.wait_for(:output, /#{p}/i)
          mflmcmd << i
        end
      end
      mflmcmd.wait_for(:exit)
    end
    only_if { ::File.exist?(::File.join(license_manager_path, 'mflmcmd')) }
    action :nothing
  end

  # license manager startup script
  template '/etc/mflmrcscript' do
    source 'etc/mflmrcscript.erb'
    cookbook 'microfocus'
    mode '0755'
    owner 'root'
    group 'root'
    variables(license_manager_path: license_manager_path)
  end

  # license manager systemd service
  systemd_service 'mflm' do
    description 'Micro Focus License Manager'
    install do
      wanted_by 'multi-user.target'
    end
    service do
      exec_start '/etc/mflmrcscript'
      exec_stop ::File.join(license_manager_path, 'lmfgetpv k')
      type 'forking'
    end
    only_if { IO.read('/proc/1/comm').chomp == 'systemd' }
  end

  # license manager init.d script
  template '/etc/init.d/mflm' do
    source 'etc/init.d/mflm.erb'
    cookbook 'microfocus'
    mode '0755'
    owner 'root'
    group 'root'
    variables(license_manager_path: license_manager_path)
    only_if { IO.read('/proc/1/comm').chomp != 'systemd' }
  end

  # license manager service
  service 'mflm' do
    supports restart: true, status: true
    action %i[enable start]
  end
end


files/default/plugins/gcc_version.rb
# Author:: Dan Webb (<d.webb@derby.ac.uk>)
# Cookbook Name:: microfocus-server
# Recipe::  gcc_version
#
# Copyright 2014 University of Derby
#
# Licensed under the Apache License, Version 2.0 (the 'License');
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an 'AS IS' BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

Ohai.plugin(:GCCVersion) do
  provides 'gcc', 'gcc/version' 
  depends 'platform'

  collect_data(:default) do
    if %w{ rhel }.include?(platform_family) or %w{ redhat centos }.include?(platform)
      gcc Mash.new
      gcc[:version] = Mixlib::ShellOut.new('gcc -dumpversion').run_command.stdout.delete!("\n")
    end
  end
end


recipes/default.rb
# Author:: Luke Bradbury (<luke.bradbury@derby.ac.uk>)
# Cookbook Name:: microfocus-server
# Recipe:: default
#
# Copyright 2013 University of Derby
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

serial_number = node['microfocus']['license']['serial number']
license_number = node['microfocus']['license']['license number']
mf_home     = node['microfocus']['server express']['home']
lmf_home    = node['microfocus']['license manager']['home']
tmp_dir     = Chef::Config['file_cache_path']
filename    = node['microfocus']['server express']['url'].split('/')[-1]
install_file = File.join(tmp_dir, filename)

chef_gem 'greenletters'
require 'greenletters'

node['microfocus']['packages'].each do |name|
  name += '.i386' if node['platform_version'].to_i == 5 && _32_bit?
  yum_package name
end

yum_package 'glibc'
yum_package 'glibc.i686'
yum_package 'libgcc'
yum_package 'libgcc.i686'
yum_package 'libstdc++'

directory mf_home do
  recursive true
end

directory lmf_home do
  recursive true
end

remote_file install_file do
  source node['microfocus']['server express']['url']
end

ruby_block 'install' do
  block do
    install = Greenletters::Process.new("#{mf_home}/install", transcript: $stdout, timeout: 300)
    install.on(:output, /--more--/i) do
      install << ' '
    end

    install.start!
    # 12B-1-2.
    # 9 continue?
    install.wait_for(:output, /(y\/n)/i)
    install << "y\n"
    # 10 license?
    install.wait_for(:output, /(y\/n)/i)
    install << "y\n"
    # 11 not standard install?
    install.wait_for(:output, /(y\/n)/i)
    install << "y\n"
    # 12 confirm understanding?
    install << "\n"
    install.wait_for(:output, /(y\/n)/i)
    install << "y\n"
    # 13 java?
    install.wait_for(:output, /(y\/n)/i)
    install << "n\n"
    # 14 lmf?
    install.wait_for(:output, /(y\/n)/i)
    install << "y\n"
    # 15 lmf dir
    install.wait_for(:output, /mflmf/i)
    install << "#{lmf_home}\n"
    install.wait_for(:output, /(y\/n)/i)
    install << "y\n"

    # 16 superuser?
    install.wait_for(:output, /(y\/n)/i)
    install << "y\n"

    # 17 autostart?
    # install.wait_for(:output, /(y\/n)/i)
    # install << "y\n"
    # 18 notice only
    # 19 architecture?
    install.wait_for(:output, /set the system default mode:/i)
    install << "64\n"

    # 20 Enterprise Server?
    install.wait_for(:output, /(y\/n)/i)
    install << "n\n"

    # 21 XDB?
    install.wait_for(:output, /(y\/n)/i)
    install << "n\n"

    install.wait_for(:exit)
  end
  action :nothing
end

ruby_block 'license' do
  block do
    license = Greenletters::Process.new("cd #{lmf_home};./mflmcmd", transcript: $stdout, timeout: 300)
    license.start!
    license.wait_for(:output, / /i)
    license << "I\n"
    license << "#{serial_number}\n"
    license << "#{license_number}\n"
    license.wait_for(:exit)
  end
  action :nothing
  notifies :run, 'execute[start license manager]'
end

execute 'stop license manager' do
  cwd "#{lmf_home}"
  command "#{lmf_home}/lmfgetpv k"
  action :nothing
  ignore_failure true
end

execute 'start license manager' do
  cwd "#{lmf_home}"
  command "#{lmf_home}/mflmman"
  action :nothing
end

execute 'extract' do
  command "tar xvf #{install_file} -C #{mf_home}"
  not_if { ::File.exist?("#{mf_home}/install") }
  notifies :run, resources(ruby_block: 'install')
  notifies :run, resources(ruby_block: 'license')
end

node.default['ohai']['plugins']['microfocus-server'] = 'plugins'
include_recipe 'ohai::default'

template 'cobopt' do
  mode 00444
  variables(
    gcc_version: node['gcc']['version']
  )
  path ::File.join(mf_home, '/etc/cobopt')
end

template 'cobopt64' do
  mode 00444
  variables(
    gcc_version: node['gcc']['version']
  )
  path ::File.join(mf_home, '/etc/cobopt64')
end
